# Musa
This repository contains the scripts in bash and R used in chapters 2 and 3 of the thesis: Species-wide diversity of Musa spp. NLR genes and their distinct response to Fusarium TR4 infection submited the 30th of September of 2025

## üóÇÔ∏è Table of Contents
- [Musa diversity genotypic characterization](#1-musa-diversity-genotypic-characterization)
- [Extraction of SNP NLRs](#2-Extraction-of-SNP-NLRs)
- [De Novo Assemblies](#3-De-Novo-Assemblies)
- [Annotation Pipeline](#4-Annotation-Pipeline)
- [NLR Ortholog clustering and NLR arquictecture](#5-NLR-Ortholog-clustering-and-NLR-arquictecture)
- [Musa NLRome Characterization](#6-Musa-NLRome-Characterization)

***
# 1. Musa diversity genotypic characterization  

A diversity panel of 201 accesions was genotyped with a SNP-based variant calling used for principal component anaylsis (pca), phylogeny and population structure. The reference genome of Musa acuminata spp malaccensis (DH-Pahang) v4 (AA_v4) published in 2021 (https://www.nature.com/articles/s42003-021-02559-3) was used to align the reads of the accesions in the panel for the variant calling.

Raw reads of were trimmed with Trim Galore v0.5 to remove the adaptors and bases with quality <20. Fastqc v0.11.9 was sourced inside Trim galore to clip 5 bp on the 5‚Äô and 3‚Äô ends after review of the read quality (0_trimgalore.sh). The resulting read files in FASTQ format were aligned to the indexed AA_v4. 

```
# Index of reference genome: run in the console of shell
source bwa-0.7.13
sbatch --wrap="bwa index ../genome_AA_v4/Musa_acuminata_pahang_v4.fasta"

#run for all the samples in scratch
for line in $(cat musa_trimmed_symlinks/sample_pagenome_list1.txt)
do sbatch run_bwa_trimed.sh $line
done

```

The resulting BAM files were generated with BWA MEM v0.7.13 with the options -M and -R to be compatible with GATK and BCFtools, defining read groups for the variant calling (2_run_bwa_trimed.sh). 

## SNP Calling

Sorting and duplicated reads identification was performed with Samtools sort v1.7 with the -b -S -h options and Picard tools v2.1.1 MarkDuplicates (3_run_sorted_dedu.sh). The M. acuminata cv. Pahang v4 (AA_v4) reference was indexed with samtools dict and faidx (4_run_AA_index.sh). All sorted bam files were indexed with Bamtools v2.4.0 (5_run_mdup_index.sh). SNP calling for the ‚Äúpangenome reads‚Äù samples was done by chromosome using all bam files as input. The variant call was done using GATK HaplotypeCaller v4.2.0.0 and the final variant call files (VCF) were merged per chromosome with the VCF per chromosome generated by Janet Higgins, also with the option ‚Äú‚Äîploidy 3‚Äù (6_run_gatk4_bychr.sh). The merge was performed with GATK v3.6.0 with option ‚Äú-T CombineVariants‚Äù (7_run_mergeVCFs_JH_ACO.sh). A recoding of the dosage calls to ploidy 2 was done, due to the compatibility restrictions by the population structure software (8_run_bcftools.sh). The resulting VCF files were filtered with BCFtools filter v.1.10.2 for a minimum quality value of 100, depth read between 10 to 300 reads (this file was later used for the SNP NLR extraction, and for biallelic SNPs. GATK SelectVariants v4.2.0.0 removed the sites with more than 30% of missing samples (9_run_filtering.sh). The last filter was for minor allele frequency (MAF) of 0.1% using BCFtools and thinning to retain only one SNP within each 10 bp window (--thin 10). A single VCF call file was created concatenating all chromosome VCFs using BCFtools concat v1.15.1_CBG (10_run_concat.sh).

## Population Structure

The populations structure was estimaed with STRUCTURE v2.3.5, the filtered VCF file was thinned further with BCFtools, keeping one SNP every 50 bp (--thin 50), as there were constraints in the total number of SNPs for this software. The admixture model was run with a burn-in period of 10,000, with 50,000 MCMC iterations and a maximum of ten founder populations (K 2 to 10). Twenty independent runs for each population K were simulated and ten random replicated Q-matrixes were aligned with POPHELPER in R. A delta K (ŒîK) was calculated using the Evanno method in POPHELPER to identify the most likely number of clusters in the panel. The filtered VCF was also used for principal component analysis (PCA) using Tassel v5.2.96. The R package ggplot2 was used to plot the PCA results. The packages ggfortify and concaveman were used to improve the information and aesthetics of the PCA. The phylogenetic tree was built with SNPrelate in R using the functions identity-by-descent (snpgfdsIBD) and hierarchical clustering (snpgdsHCluster) and the results were plotted using iTOL.

# 2. Extraction of SNP NLRs  

# 3. De Novo Assemblies  

# 4. Annotation Pipeline

<img width="4927" height="2429" alt="image" src="https://github.com/user-attachments/assets/5901f554-9c72-4f78-a98c-365e6e405478" />


# 5. NLR Ortholog clustering and NLR arquictecture 

# 6. Musa NLRome Characterization 


